d3a_interface/device_validator.py

```diff
@@ -155,85 +155,85 @@ def validate_pv_device(**kwargs):
 
 def validate_storage_device(**kwargs):
     if ("initial_soc" in kwargs and kwargs["initial_soc"] is not None) and not \
-            StorageSettings.INITIAL_CHARGE_RANGE.min <= kwargs["initial_soc"] <= \
-            StorageSettings.INITIAL_CHARGE_RANGE.max:
+            (StorageSettings.INITIAL_CHARGE_RANGE.min <= kwargs["initial_soc"] <=
+             StorageSettings.INITIAL_CHARGE_RANGE.max):
         raise D3ADeviceException(
             {"mis_configuration": [f"initial_soc should be in between "
                                    f"{StorageSettings.INITIAL_CHARGE_RANGE.min} & "
                                    f"{StorageSettings.INITIAL_CHARGE_RANGE.max}."]})
     if ("min_allowed_soc" in kwargs and kwargs["min_allowed_soc"] is not None) and not \
-            StorageSettings.MIN_SOC_RANGE.min <= kwargs["min_allowed_soc"] <= \
-            StorageSettings.MIN_SOC_RANGE.max:
+            (StorageSettings.MIN_SOC_RANGE.min <= kwargs["min_allowed_soc"] <=
+             StorageSettings.MIN_SOC_RANGE.max):
         raise D3ADeviceException(
             {"mis_configuration": [f"min_allowed_soc should be in between "
                                    f"{StorageSettings.MIN_SOC_RANGE.min} & "
                                    f"{StorageSettings.MIN_SOC_RANGE.max}."]})
     if ("initial_soc" in kwargs and kwargs["initial_soc"] is not None) and \
             ("min_allowed_soc" in kwargs and kwargs["min_allowed_soc"] is not None) and \
-            kwargs["initial_soc"] < kwargs["min_allowed_soc"]:
+            (kwargs["initial_soc"] < kwargs["min_allowed_soc"]):
```

Comment:
Parentheses here do not hurt, therefore it is ok as it is, but I would like to point out that they are not really needed, because the `and` operator has lower precedence than `<` and comparisons in general. Better safe than sorry though, leave it as it is.